### PR TITLE
Alert subscriptions - Part 1 - no actual sending or SMSing unless admin user clicks button

### PR DESCRIPTION
**Pull request by jamesjefferies**

### CODE REVIEW

File: app/services/alerts/generate_subscription_events.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    def perform
+      @school.alerts.latest.each do |alert|
+        process_alert(alert)
+      end
+    end
+
+  private
+
+    def process_alert(alert)
+      if any_subscriptions?(@school, alert.alert_type)
```
Comment (jamesalmond) [inline app/services/alerts/generate_subscription_events.rb]: Not sure this guard is necessary. If there are no subscriptions the `each` block in process subscriptions won't run which is effectively the same thing?

File: app/views/alert_mailer/alert_email.html.erb [hunk 2]
```diff
@@ -1,27 +1,27 @@
 <h1>Energy Sparks alerts for <%= link_to @school.name, school_url(@school) %></h1>
 <p>You have been assigned as a contact for the following alerts from Energy Sparks.</p>
 <% if @school.school_admin.any? %>
```
Comment (jamesalmond) [inline app/views/alert_mailer/alert_email.html.erb]: Could this just be a check for some unsubscribe emails? This way there's logic for them being from the school admin in two places.

File: app/services/alerts/generate_subscription_events.rb (hunk 1 above)
Comment (jamesjefferies) [reply app/services/alerts/generate_subscription_events.rb]: Yep - good spot

File: app/views/alert_mailer/alert_email.html.erb (hunk 2 above)
Comment (jamesjefferies) [reply app/views/alert_mailer/alert_email.html.erb]: Yep, also good spot - best to keep school.school_admin out of it..

**Pull request merged**